samples/cpp/speech_sample/main.cpp

```
@@ -115,16 +115,36 @@ int main(int argc, char* argv[]) {
                 }
             }
             check_number_of_inputs(model->inputs().size(), numInputFiles);
-            const ov::Layout tensor_layout{"NC"};
             ov::preprocess::PrePostProcessor proc(model);
-            for (int i = 0; i < model->inputs().size(); i++) {
-                proc.input(i).tensor().set_element_type(ov::element::f32).set_layout(tensor_layout);
+            const auto& inputs = model->inputs();
+            std::map<std::string, std::string> custom_layouts;
+            if (!FLAGS_layout.empty()) {
+                custom_layouts = parse_input_layouts(FLAGS_layout, inputs);
+            }
+            for (const auto& input : inputs) {
+                const auto& item_name = input.get_any_name();
+                auto& in = proc.input(item_name);
+                in.tensor().set_element_type(ov::element::f32);
+                // Explicitly set inputs layout
+                if (custom_layouts.count(item_name) > 0) {
+                    in.model().set_layout(ov::Layout(custom_layouts.at(item_name)));
+                }
             }
             for (int i = 0; i < model->outputs().size(); i++) {
                 proc.output(i).tensor().set_element_type(ov::element::f32);
             }
             model = proc.build();
-            ov::set_batch(model, batchSize);
+            if (FLAGS_bs) {
+                if (FLAGS_layout.empty() &&
+                    std::any_of(inputs.begin(), inputs.end(), [](const ov::Output<ov::Node>& i) {
+                        return ov::layout::get_layout(i).empty();
+                    })) {
+                    slog::err << "Layout is not set for any input, so custom batch size is not set." << slog::endl;
+                    batchSize = 1;
```

We need to raise exception here instead of setting batch size to 1, we can't just ignore intention of user to set batch size

---

We set batch size variable to default value because ov::set_batch wasn't called. But it's only error for user and sample works good with default value. Why is it exception? Layout parameter is optional

---

@mgrdv User specify batch size in command lint, but don't specify where this batch size is located in input. We can't set batch size. But we should not ignore it.

---

@nkogteva , do you suggest throwing an exception and aborting work of the sample in this case? I think that we should only inform user

---

@mgrdv Yes, that exactly what I'm saying. Also message should contain hint what should be added into the command line (layout) to get it working.

---

IMHO we should throw an error here and stop execution with an understandable error message on how to fix the problem, then the user will be able to remove -bs option and run the sample successfully

---

I agree with @mryzhov  in this case. I think so

---

@nkogteva , do you have any suggestions about correct text message with hint what should be added into the command line (layout)?

---

@mgrdv Something like this:
```
"Can't set batch size to" + FLAGS_bs + " since input layouts are not specified. Please use command line parameter --layout to specify where batch size located. For example, input1[NCHW],input2[NC] or [NC]"
```

---

good, I will change

---

@nkogteva , I added it